client/core/trade.go

```diff
@@ -1722,38 +1761,61 @@ func (t *trackedTrade) processRedemption(msgID uint64, redemption *msgjson.Redem
 	// Update the database.
 	auth.RedemptionSig = redemption.Sig
 	auth.RedemptionStamp = redemption.Time
-	redeemAsset := t.wallets.fromAsset // this redeem is the other party's
+
 	if dbMatch.Side == order.Maker {
-		// You are the maker, this is the taker's redeem.
-		log.Debugf("Notified of taker's redemption (%s: %v) for order %v...",
-			redeemAsset.Symbol, coinIDString(redeemAsset.ID, redemption.CoinID), t.ID())
-		match.SetStatus(order.MatchComplete)
-		proof.TakerRedeem = []byte(redemption.CoinID)
-		err := t.db.UpdateMatch(&match.MetaMatch)
-		if err != nil {
-			return errs.add("error storing match info in database: %v", err)
+		if err := t.processTakersRedemption(match, redemption.CoinID); err != nil {
+			errs.addErr(err)
+		}
+	} else {
+		if err := t.processMakersRedemption(match, redemption.CoinID, redemption.Secret); err != nil {
+			errs.addErr(err)
 		}
-		return nil
 	}
 
-	// You are the taker, this is the maker's redeem. The maker is the
-	// initiator, so validate the revealed secret.
+	err = t.db.UpdateMatch(&match.MetaMatch)
+	if err != nil {
+		errs.add("error storing match info in database: %v", err)
+	}
```

Comment:
This appears redundant with the `UpdateMatch` in either `processTakersRedemption` or `processMakersRedemption`.

Reply:
There is no `UpdateMatch` in `processMakersRedemption`, but I meant to remove it from `processTakersRedemption` too. Nice catch.